crates/autopilot/src/run_loop.rs

```diff
@@ -621,28 +756,36 @@ impl RunLoop {
     /// transaction has been mined.
     async fn settle(
         &self,
-        driver: &infra::Driver,
-        solved: &competition::SolutionWithId,
+        driver: Arc<infra::Driver>,
+        solution_id: u64,
+        solved_order_uids: HashSet<OrderUid>,
         auction_id: i64,
         submission_deadline_latest_block: u64,
     ) -> Result<(), SettleError> {
         let request = settle::Request {
-            solution_id: solved.id(),
+            solution_id,
             submission_deadline_latest_block,
         };
         let tx_hash = self
-            .wait_for_settlement(driver, auction_id, request)
+            .wait_for_settlement(driver.clone(), auction_id, request)
             .await?;
         tracing::debug!(?tx_hash, "solution settled");
 
+        self.in_flight_orders
+            .lock()
+            .await
+            .retain(|order| !solved_order_uids.contains(order));
+
+        self.settlement_queues.lock().await.remove(&driver.name);
```

Comment:
Not sure why the submission queue for that particular solver should be removed here. I thought the cleanup mechanism (which I don't think is necessary in the first place) was supposed to be the task that consumes the queue and settles the orders in the background.

Reply:
I forgot to remove it actually.